central/image/datastore/store/v2/postgres/store.go

```diff
@@ -958,3 +960,22 @@ func getCVEComponentIndex(s string) int {
 	}
 	return index
 }
+
+func (s *storeImpl) componentsTableEmpty(ctx context.Context, obj *storage.Image) bool {
+	conn, release, err := s.acquireConn(ctx, ops.Get, "Image")
+	if err != nil {
+		return false
+	}
+	defer release()
+
+	tx, err := conn.Begin(ctx)
+	if err != nil {
+		return false
+	}
+
+	if components, err := getImageComponents(ctx, tx, obj.GetId()); err == nil && len(components) == 0 {
```

Comment:
We should probably return the err back and return it back to the callers.  If we error out we probably have other issues.